src/main/java/emu/grasscutter/command/commands/ClearCommand.java

```diff
@@ -61,43 +69,84 @@ public void execute(Player sender, Player targetPlayer, List<String> args) {
-                CommandHandler.sendTranslatedMessage(sender, "commands.clear.weapons", playerString);
-                playerInventory.removeItems(
+
+        String id = args.remove(0);
+
+        try {
+            switch (id) {
+                case "wp" -> {
+                    if (param.amount != -1) {
+                        playerInventory.removeItems(getWeapons(playerInventory, param).toList().subList(0, param.amount));
+                        CommandHandler.sendTranslatedMessage(sender, "commands.clear.weapons", playerString, param.amount);
+                    } else {
+                        playerInventory.removeItems(getWeapons(playerInventory, param).toList());
+                        CommandHandler.sendTranslatedMessage(sender, "commands.clear.weapons_all", playerString);
+                    }
+
+                }
+                case "art" -> {
+                    if (param.amount != -1) {
+                        playerInventory.removeItems(getRelics(playerInventory, param).toList().subList(0, param.amount));
+                        CommandHandler.sendTranslatedMessage(sender, "commands.clear.artifacts", playerString, param.amount);
+                    } else {
+                        playerInventory.removeItems(getRelics(playerInventory, param).toList());
+                        CommandHandler.sendTranslatedMessage(sender, "commands.clear.artifacts_all", playerString);
+                    }
+                }
+                case "mat" -> {
+                    if (param.amount != -1) {
+                        playerInventory.removeItems(getOther(ItemType.ITEM_MATERIAL, playerInventory, param).toList().subList(0, param.amount));
+                        CommandHandler.sendTranslatedMessage(sender, "commands.clear.materials", playerString, param.amount);
+                    } else {
+                        playerInventory.removeItems(getOther(ItemType.ITEM_MATERIAL, playerInventory, param).toList());
+                        CommandHandler.sendTranslatedMessage(sender, "commands.clear.materials_all", playerString);
+                    }
+                }
+                case "all" -> {
+                    playerInventory.removeItems(getRelics(playerInventory, param).toList());
+                    CommandHandler.sendTranslatedMessage(sender, "commands.clear.artifacts", playerString);
+                    playerInventory.removeItems(getWeapons(playerInventory, param).toList());
+                    CommandHandler.sendTranslatedMessage(sender, "commands.clear.weapons", playerString);
+                    playerInventory.removeItems(
                         getOther(ItemType.ITEM_MATERIAL, playerInventory, param).toList());
-                CommandHandler.sendTranslatedMessage(sender, "commands.clear.materials", playerString);
-                playerInventory.removeItems(
+                    CommandHandler.sendTranslatedMessage(sender, "commands.clear.materials", playerString);
```

Comment:
I think what you want is the `*_all` styled message?

```suggestion
                    CommandHandler.sendTranslatedMessage(sender, "commands.clear.artifacts_all", playerString);
                    playerInventory.removeItems(getWeapons(playerInventory, param).toList());
                    CommandHandler.sendTranslatedMessage(sender, "commands.clear.weapons_all", playerString);
                    playerInventory.removeItems(
                        getOther(ItemType.ITEM_MATERIAL, playerInventory, param).toList());
                    CommandHandler.sendTranslatedMessage(sender, "commands.clear.materials_all", playerString);
```

Reply:
Sorry, I forgot to modify all mode